src/storage/src/types/connections.rs

```diff
@@ -219,6 +218,7 @@ impl ConfigKey for KafkaConnectionOptionName {
         use KafkaConnectionOptionName::*;
         match self {
             Broker | Brokers => "bootstrap.servers",
+            ProgressTopic => unimplemented!("not a kafka config"),
```

Comment:
It's a tad unfortunately that `ProgressTopic` is part of this enum given that it's not actually a connection option and panics here.

How much follow-up work do you think it would take to split this from the other options that are in fact kafka configs as opposed to configs about our own implementation of the connection object? (this is something that I could volunteer to work on).  The `KafkaConnectionOptionName` works is fairly new work and I'd hate to pollute it with panics like this so soon!

Reply:
This might make sense to tackle as part of: https://github.com/MaterializeInc/materialize/issues/14656

My understanding is that medium term we don't expect connection options to be 1:1 with rdkafka config keys, so hopefully this panic doesn't live too long either. I don't think it would be too hard to rewrite this without the `ConfigKey` abstraction, which is maybe a closer-term solution to the problem...